internal/characters/cyno/attack.go

```diff
@@ -0,0 +1,138 @@
+package cyno
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+)
+
+var (
+	attackFrames          [][]int
+	attackHitmarks        = [][]int{{14}, {17}, {13, 22}, {27}}
+	attackHitlagHaltFrame = [][]float64{{0.01}, {0.06}, {0, 0.02}, {0.04}}
+	attackDefHalt         = [][]bool{{false}, {true}, {false, true}, {true}}
+)
+
+const normalHitNum = 4
+
+func init() {
+	attackFrames = make([][]int, normalHitNum)
+	attackFrames[0] = frames.InitNormalCancelSlice(attackHitmarks[0][0], 28) // N1 -> N2
+	attackFrames[0][action.ActionAttack] = 15
+
+	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1][0], 23) // N2 -> N3
+	attackFrames[1][action.ActionAttack] = 22
+
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2][1], 27) // N3 -> N4
+	attackFrames[2][action.ActionCharge] = 26
+
+	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3][0], 58) // N4 -> N1
+	attackFrames[3][action.ActionCharge] = 500                               // impossible action
+}
+
+func (c *char) Attack(p map[string]int) action.ActionInfo {
+	if c.StatusIsActive(burstKey) {
+		c.NormalHitNum = burstHitNum
+		return c.attackB(p) // go to burst mode attacks
+	}
+	if c.NormalHitNum >= burstHitNum { // this should avoid the panic error
+		c.NormalHitNum = normalHitNum
+		c.ResetNormalCounter()
+	}
+	c.NormalHitNum = normalHitNum
+	for i, mult := range attack[c.NormalCounter] {
+		ai := combat.AttackInfo{
+			ActorIndex:         c.Index,
+			Abil:               fmt.Sprintf("Normal %v", c.NormalCounter),
+			Mult:               mult[c.TalentLvlAttack()],
+			AttackTag:          combat.AttackTagNormal,
+			ICDTag:             combat.ICDTagNormalAttack,
+			ICDGroup:           combat.ICDGroupDefault,
+			Element:            attributes.Physical,
+			Durability:         25,
+			HitlagFactor:       0.01,
+			HitlagHaltFrames:   attackHitlagHaltFrame[c.NormalCounter][i],
+			CanBeDefenseHalted: attackDefHalt[c.NormalCounter][i],
+		}
+
+		c.Core.QueueAttack(
+			ai,
+			combat.NewCircleHit(c.Core.Combat.Player(), 0.5, false, combat.TargettableEnemy),
```

Comment:
actual hitbox info is:
{{radius 1.8 circle}, {x=1.8 z=2.7 box}, {x=2.2 z=3.6 box, x=2.2 z=3.6 box}, {radius 2.3 circle}}